src/main/java/com/itechart/javalab/library/dao/impl/SqlBookDao.java

```diff
@@ -66,7 +69,40 @@ public static BookDao getInstance() {
         return Optional.of(books);
     }
 
+    @Override
+    public Optional<Integer> getCountOfBooksId(boolean isFiltered) {
+
+        int countBooksRecords = 0;
+
+        String query = addAvailableConditions(GET_COUNT_BOOKS_RECORDS, isFiltered);
+
+        try (Connection connection = connectionPool.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                countBooksRecords = resultSet.getInt(1);
+            }
+
+        } catch (SQLException e) {
+            log.error("SqlException in attempt to get Connection", e);
+            return Optional.empty();
+        }
+        return Optional.of(countBooksRecords);
+    }
 
 
+    private String addAvailableConditions(String query, boolean isFiltered) {
+        return isFiltered ? query + " WHERE in_stock > 0" : query;
```

Comment:
do not concatenate strings in frequently called methods, this will cause performance problems, and `query` is always a constant.